### DISCUSSION TITLE
Zones

### DISCUSSION THREAD

**Discussion by jay-oswald** [Freika/dawarich#75] [Ideas]
I would love the ability to set zones. So say mark my home with a 300m radius as home, mark the shops, mark work etc.

Then history could show the raw location history but also zone history.

E.g. Arrived at work at 9:03am, left work at 5:04pm, arrived home at 5:34pm etc.

Allowing a quick text based overview of location history. 

I would also the like zone data to be added to the future API. Then I would like to do a Home Assistant Integration, and include the zone data in the API and then in Home Assistant. This would allow more advanced automations than HA allows with its limited location data. I'm thinking show current zone and previous zone.

It would allow automations like if current zone is shops and previous zone was work, HA could send me the quick shop shopping list. But if previous zone was home send me the full shopping list etc.

Also exposing zone text history to Home Assistant would allow integrations with voice assistants. So could ask "where is my wife" and it could respond "she left work 14 mins ago".

Zones plus the already planned API just open up a lot of potential

**Comment by Freika** [comment]
I love the idea of zones too, and I'll certainly consider adding it in the future, although I can't promise any kind of automation support for now. It might be implemented later in the form of webhooks (I'm home, send a webhook) or some other form.

**Reply by jay-oswald** [reply]
I wouldn't expect any form of automation, just use the API or webhooks to get updates into HA and leave the rest up to that.

Ideally I think the best result would be a webhook that pushed data to HA every time there was an update, as that would get the fastest updates in HA without a high polling rate for an API
